lib/CL/clEnqueueCommandBufferKHR.c

```diff
@@ -115,14 +115,39 @@ POname (clEnqueueCommandBufferKHR) (cl_uint num_queues,
   POCL_UNLOCK (command_buffer->mutex);
   POCL_RETURN_ERROR_COND ((!is_ready), CL_INVALID_OPERATION);
 
-  /* Submit to queue(s) */
-  if (num_used_queues == 1 && used_queues[0]->device->ops->run_command_buffer)
+  /* Submit to queue(s) directly if device supports cmdbufs natively */
+  if (num_used_queues == 1 && used_queues[0]->device->native_command_buffers)
     {
       /* TODO: add base event id & increment global event id counter by number
        * of commands generated by the buffer */
-      return used_queues[0]->device->ops->run_command_buffer (
-        used_queues[0]->device, command_buffer);
+      /* TODO: figure out a way to handle migrations effectively. We can't
+       * change the cmd buffer once it's finished recording, as changing would
+       * also require changing the hardware resource (e.g. L0 command list),
+       * which would negate the point of using cmdbufs in the 1st place. */
+      _cl_command_node *node = NULL;
+      cl_event final_ev;
+      errcode = pocl_create_command (
+        &node, used_queues[0], CL_COMMAND_COMMAND_BUFFER_KHR, &final_ev,
+        num_events_in_wait_list, event_wait_list, NULL);
+      // pocl_deep_copy_migration_info_list (cmd->migr_infos, 0));
```

Comment:
Is this omitted by accident? Can be removed or at least commented to a C comment?

Reply:
Converted to `#if 0`